src/main/java/io/coala/jetbrains/utils/deserializers/CodeAnalysisLogDeserializer.java

```diff
@@ -0,0 +1,62 @@
+package io.coala.jetbrains.utils.deserializers;
+
+import gherkin.deps.com.google.gson.GsonBuilder;
+import gherkin.deps.com.google.gson.JsonArray;
+import gherkin.deps.com.google.gson.JsonDeserializationContext;
+import gherkin.deps.com.google.gson.JsonDeserializer;
+import gherkin.deps.com.google.gson.JsonElement;
+import gherkin.deps.com.google.gson.JsonObject;
+import gherkin.deps.com.google.gson.JsonParseException;
+import gherkin.deps.com.google.gson.reflect.TypeToken;
+import io.coala.jetbrains.utils.CodeAnalysisLog;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CodeAnalysisLogDeserializer {
+
+  /**
+   * This method deserializes raw JSON log output from coala analysis
+   * and converts it to Java friendly objects.
+   *
+   * @param jsonString the String to be deserialized
+   * @return list of {@link CodeAnalysisLog} objects
+   */
+  public static List<CodeAnalysisLog> getAllCodeAnalysisLogs(String jsonString) {
+    final GsonBuilder gsonBuilder = new GsonBuilder();
+    gsonBuilder.registerTypeAdapter(List.class, new ExtractCodeAnalysisLogsFromJson());
+    Type typeOfListOfCodeAnalysisLogs = new TypeToken<List<CodeAnalysisLog>>() {}.getType();
+    return gsonBuilder.create().fromJson(jsonString, typeOfListOfCodeAnalysisLogs);
+  }
+
+  /**
+   * This method creates a new log object from the corresponding JsonElement.
+   *
+   * @param jsonElement the element to be parsed
+   * @return new {@link CodeAnalysisLog} object synthesized from JSON
+   */
+  public static CodeAnalysisLog getNewCodeAnalysisLog(JsonElement jsonElement) {
+    final GsonBuilder gsonBuilder = new GsonBuilder();
+    gsonBuilder.registerTypeAdapter(CodeAnalysisLog.class, new DeserializeLogFromJson());
+    return gsonBuilder.create().fromJson(jsonElement, CodeAnalysisLog.class);
+  }
+
+  private static class ExtractCodeAnalysisLogsFromJson implements
+      JsonDeserializer<List<CodeAnalysisLog>> {
+
+    @Override
+    public List<CodeAnalysisLog> deserialize(JsonElement jsonElement, Type type,
+        JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
+      final JsonObject jsonObject = jsonElement.getAsJsonObject();
+      final JsonArray logsList = jsonObject.get("logs").getAsJsonArray();
```

Comment:
`logs` literal shoul preferably be a static property.

Reply:
I don't see any benefit of making this `static`. This is just an intermediate list of raw json logs which are processed in the next step.

Reply:
Those kinds of suggestions are generally meant to maintain immunity towards API breakages. If you end assuming the literal in multiple places, you'll have to update them all which isn't desireable.

Reply:
That's right. However, this is the only place I've used `"logs"` since I need to access the property directly from the JSON string. If coala decides to change this in the future, on the string at a single needs to be changed.